metals/src/main/scala/scala/meta/internal/metals/Compilers.scala

```diff
@@ -666,6 +711,76 @@ class Compilers(
       (log ++ filteredOptions).asJava
     )
   }
+
+  private def toDebugCompletionType(
+      kind: CompletionItemKind
+  ): CompletionItemType = {
+    kind match {
+      case CompletionItemKind.Constant => CompletionItemType.VALUE
+      case CompletionItemKind.Value => CompletionItemType.VALUE
+      case CompletionItemKind.Keyword => CompletionItemType.KEYWORD
+      case CompletionItemKind.Class => CompletionItemType.CLASS
+      case CompletionItemKind.TypeParameter => CompletionItemType.CLASS
+      case CompletionItemKind.Operator => CompletionItemType.FUNCTION
+      case CompletionItemKind.Field => CompletionItemType.FIELD
+      case CompletionItemKind.Method => CompletionItemType.METHOD
+      case CompletionItemKind.Unit => CompletionItemType.UNIT
+      case CompletionItemKind.Enum => CompletionItemType.ENUM
+      case CompletionItemKind.Interface => CompletionItemType.INTERFACE
+      case CompletionItemKind.Constructor => CompletionItemType.CONSTRUCTOR
+      case CompletionItemKind.Folder => CompletionItemType.FILE
+      case CompletionItemKind.Module => CompletionItemType.MODULE
+      case CompletionItemKind.EnumMember => CompletionItemType.ENUM
+      case CompletionItemKind.Snippet => CompletionItemType.SNIPPET
+      case CompletionItemKind.Function => CompletionItemType.FUNCTION
+      case CompletionItemKind.Color => CompletionItemType.COLOR
+      case CompletionItemKind.Text => CompletionItemType.TEXT
+      case CompletionItemKind.Property => CompletionItemType.PROPERTY
+      case CompletionItemKind.Reference => CompletionItemType.REFERENCE
+      case CompletionItemKind.Variable => CompletionItemType.VARIABLE
+      case CompletionItemKind.Struct => CompletionItemType.MODULE
+      case CompletionItemKind.File => CompletionItemType.FILE
+      case _ => CompletionItemType.TEXT
+    }
+  }
+
+  private def toDebugCompletionItem(
+      item: CompletionItem,
+      adjustColumn: Int,
+      expressionLength: Int
+  ): DebugCompletionItem = {
+    val debugItem = new DebugCompletionItem()
+    debugItem.setLabel(item.getLabel())
+    val edit = item.getTextEdit()
+    val (newText, range) = if (edit.isLeft()) {
+      (edit.getLeft().getNewText(), edit.getLeft().getRange())
+    } else {
+      (edit.getRight().getNewText(), edit.getRight().getInsert())
+    }
+    val start = range.getStart().getCharacter() - adjustColumn
+    val end = range.getEnd().getCharacter() - adjustColumn
+
+    val length = end - start
+    debugItem.setLength(length)
+
+    // remove snippets, since they are not supported in DAP
+    val fullText = newText.replaceAll("\\$[1-9]+", "")
```

Comment:
Is snippet a well-known name for 'parameter placeholder' (it's my private name for them) ?

Reply:
I don't know if i'd say 'parameter placehoder', but at least in the LSP sense, snippets is a well-known name for just 'placeholder' or 'tab stops', which the [docs](https://microsoft.github.io/language-server-protocol/specification#snippet_syntax) use quite a bit.

Reply:
@tgodzik wdyt about adding a parameter to `PresentationCompiler.complete` function to disable snippets?

Reply:
We do want them actually to get the end position of the cursor `$0`